### PR TITLE
CI: Build docs in CI

### LINKED ISSUE #497: Test docs build in CI
Currently, the documentation build is not tested by `pythonpackage.yml`. I suggest adding a separate workflow for docs changes only, and to run `pythonpackage.yml` for code changes only.

Moreover, consider testing multiple Python versions in the CI matrix.

### PR DESCRIPTION
**Pull request by erlend-aasland**
Resolves #497

### CODE REVIEW

File: doc/requirements.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
-sphinx
-sphinx-autodoc-typehints
-furo
+sphinx==7.3.7
+sphinx-autodoc-typehints==2.2.2
+furo==2024.5.6
```
Comment (acolomb) [inline doc/requirements.txt:3]: Why are you pinning the versions here?  At the least, I would expect to list only minimum, but not specific versions.

**Comment by acolomb** [comment]
Our "CI" for the docs actually lives on readthedocs.org, as each commit on `master` is built there under the `latest` version.  Testing here may actually lead to less PRs getting merged that break the online docs, so that's positive.

This doesn't make an effort though to separate action runs between docs-only and code-only changes, as mentioned in #497?  Since any code change also affects the docs, and docs-only fixes are rare, I can live with that.

**Comment by erlend-aasland** [comment]
> This doesn't make an effort though to separate action runs between docs-only and code-only changes, as mentioned in https://github.com/christiansandberg/canopen/issues/497? Since any code change also affects the docs, and docs-only fixes are rare, I can live with that.

Yes. I'm not used to the docstrings being so tightly coupled with the rendered docs; I simply forgot that it's not possible to do that separation with the current setup.

File: doc/requirements.txt (hunk 1 above)
Comment (erlend-aasland) [reply doc/requirements.txt:3]: Hm, yes this may be a tad too specific. Using `~=major.minor` or `~=major` would be a better option.

Comment (acolomb) [reply doc/requirements.txt:3]: I'd suggest leaving them unspecified or give only a minimum. What do we gain by forbidding newer versions?  It seems to have worked all this time.

Comment (erlend-aasland) [reply doc/requirements.txt:3]: I prefer to ping dependencies at some level, often at the minor version (assuming SemVer). Having a fairly reproducible test environment is a nice thing.

**Comment by acolomb** [comment]
I just hope Someone(TM) will be there to bump these versioned dependencies when they have bit-rotted enough ;-)

**Comment by acolomb** [comment]
So it seems there are no artifacts?  Wouldn't it make sense to build some kind of HTML bundle while we're already generating the docs?

**Comment by erlend-aasland** [comment]
> So it seems there are no artifacts? Wouldn't it make sense to build some kind of HTML bundle while we're already generating the docs?

That's a very good idea.

**Pull request merged**